spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/sqs/SqsAutoConfigurationTest.java

```diff
@@ -0,0 +1,200 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.awspring.cloud.autoconfigure.ConfiguredAwsClient;
+import io.awspring.cloud.autoconfigure.core.AwsAutoConfiguration;
+import io.awspring.cloud.autoconfigure.core.AwsClientCustomizer;
+import io.awspring.cloud.autoconfigure.core.CredentialsProviderAutoConfiguration;
+import io.awspring.cloud.autoconfigure.core.RegionProviderAutoConfiguration;
+import io.awspring.cloud.sqs.annotation.SqsListenerAnnotationBeanPostProcessor;
+import io.awspring.cloud.sqs.config.EndpointRegistrar;
+import io.awspring.cloud.sqs.config.SqsMessageListenerContainerFactory;
+import io.awspring.cloud.sqs.listener.ContainerOptions;
+import io.awspring.cloud.sqs.listener.errorhandler.AsyncErrorHandler;
+import io.awspring.cloud.sqs.listener.interceptor.AsyncMessageInterceptor;
+import java.net.URI;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.Nullable;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.util.Assert;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+import software.amazon.awssdk.services.sqs.SqsAsyncClientBuilder;
+
+/**
+ * Tests for class {@link SqsAutoConfiguration}.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ */
+class SqsAutoConfigurationTest {
+
+	private static final String CUSTOM_OBJECT_MAPPER_BEAN_NAME = "customObjectMapper";
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withPropertyValues("spring.cloud.aws.region.static:eu-west-1")
+			.withConfiguration(AutoConfigurations.of(RegionProviderAutoConfiguration.class,
+					CredentialsProviderAutoConfiguration.class, SqsAutoConfiguration.class,
+					AwsAutoConfiguration.class));
+
+	@Test
+	void sqsAutoConfigurationIsDisabled() {
+		this.contextRunner.withPropertyValues("spring.cloud.aws.sqs.enabled:false")
+				.run(context -> assertThat(context).doesNotHaveBean(SqsAsyncClient.class));
+	}
+
+	@Test
+	void sqsAutoConfigurationIsEnabled() {
+		this.contextRunner.withPropertyValues("spring.cloud.aws.sqs.enabled:true").run(context -> {
+			assertThat(context).hasSingleBean(SqsAsyncClient.class);
+			assertThat(context).hasSingleBean(SqsMessageListenerContainerFactory.class);
+			assertThat(context).hasBean("defaultSqsListenerContainerFactory");
+			assertThat(context).hasBean("sqsAsyncClient");
```

Comment:
Hmm, well, the assertion here is more about the bean names, which are also part of the public API since users have access to them.

We could remove the previous lines but they might be useful in case e.g. we wrongly change the `SqsMessageListenerContainerFactory` bean to a `SqsMessageListenerContainer` one.

Maybe I'm being overzealous?